test/positivity.lean

```diff
@@ -103,3 +104,18 @@ example {a : ℤ} (ha : a > 0) : 0 ≤ a := by positivity
 example {a : ℤ} (ha : 0 < a) : a ≥ 0 := by positivity
 
 example {a : ℤ} (ha : a > 0) : a ≥ 0 := by positivity
+
+/- ## Coercions -/
+
+open_locale nnreal
+
+example {n : ℕ} : 0 ≤ n := by positivity
+example {n : ℕ} : 0 ≤ (n : ℤ) := by positivity
+example {n : ℕ} : 0 ≤ (n : ℚ) := by positivity
+example {n : ℕ} : 0 ≤ ((n : ℚ) : ℝ) := by positivity
+example {α : Type*} [ordered_ring α] {n : ℤ} : 0 ≤ (n ^ 2 : ℤ) := by positivity
```

Comment:
`α` isn't mentioned in the statement here. This statement fails:
```
example {α : Type*} [ordered_ring α] {n : ℤ} : 0 ≤ (n ^ 2 : α) := by positivity
```

Reply:
What's the thing you're trying to test here?

Reply:
Oops, I forgot the outer coercion.